sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -154,5 +166,65 @@ public virtual void StopTestRecording()
 #endif
             Recording?.Dispose(save);
         }
+
+        protected ValueTask<Response<T>> WaitForCompletionAsync<T>(Operation<T> operation)
+        {
+            if (Mode == RecordedTestMode.Playback)
+            {
+                return operation.WaitForCompletionAsync(ZeroPollingInterval, default);
+            }
+            else
+            {
+                return operation.WaitForCompletionAsync();
+            }
+        }
+
+        protected ResourcesManagementClient GetResourceManagementClient()
+        {
+            var options = InstrumentClientOptions(new ResourcesManagementClientOptions());
+            CleanupPolicy = new ResourceGroupCleanupPolicy();
+            options.AddPolicy(CleanupPolicy, HttpPipelinePosition.PerCall);
+
+            return CreateClient<ResourcesManagementClient>(
+                TestEnvironment.SubscriptionId,
+                TestEnvironment.Credential,
+                options);
+        }
+
+        protected async Task CleanupResourceGroupsAsync()
+        {
+            if (CleanupPolicy != null && Mode != RecordedTestMode.Playback)
+            {
+                var resourceGroupsClient = new ResourcesManagementClient(
+                    TestEnvironment.SubscriptionId,
+                    TestEnvironment.Credential,
+                    new ResourcesManagementClientOptions()).ResourceGroups;
+                foreach (var resourceGroup in CleanupPolicy.ResourceGroupsCreated)
+                {
+                    await resourceGroupsClient.StartDeleteAsync(resourceGroup);
+                }
+            }
+        }
+
+        protected async Task<string> GetFirstUsableLocationAsync(ProvidersOperations providersClient, string resourceProviderNamespace, string resourceType)
```

Comment:
Shouldn't we use the location from the test environment?

Reply:
Just moved this method over from Common/ManagementTestShared.

Reply:
We are staying with parity with this PR